File: doc-examples/jdbc-example-java/src/main/java/example/oracle/AbstractBookRepository.java

package example.oracle;

import example.Book;
import io.micronaut.context.annotation.Requires;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.jdbc.runtime.JdbcOperations;
import io.micronaut.data.model.query.builder.sql.Dialect;
import jakarta.transaction.Transactional;

import java.sql.ResultSet;
import java.util.List;

@JdbcRepository(dialect = Dialect.ORACLE)
@Requires(env="oracle")
public abstract class AbstractBookRepository extends example.AbstractBookRepository {

    public AbstractBookRepository(JdbcOperations jdbcOperations) {
        super(jdbcOperations);
    }

    @Transactional
    public List<Book> findByTitle(String title) {
        String sql = "SELECT * FROM Book book WHERE book.title = ?";
        return jdbcOperations.prepareStatement(sql, statement -> {
            statement.setString(1, title);
            ResultSet resultSet = statement.executeQuery();
            return jdbcOperations.entityStream(resultSet, Book.class).toList();
        });
    }
}

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/AbstractBookRepository.kt

package example

import io.micronaut.data.annotation.Repository
import io.micronaut.data.jdbc.runtime.JdbcOperations
import io.micronaut.data.repository.kotlin.KotlinCrudRepository
import javax.transaction.Transactional

@Repository
abstract class AbstractBookRepository(private val jdbcOperations: JdbcOperations) : KotlinCrudRepository<Book, Long> {

    @Transactional
    fun findByTitle(title: String): List<Book> {
        val sql = "SELECT * FROM Book AS book WHERE book.title = ?"
        return jdbcOperations.prepareStatement(sql) { statement ->
            statement.setString(1, title)
            val resultSet = statement.executeQuery()
            jdbcOperations.entityStream(resultSet, Book::class.java)
                    .toList()
        }
    }
}
